vips/vip-Gateway/bscmainnet.ts

```diff
@@ -0,0 +1,260 @@
+    description: `
+    This VIP does the following:
+    1. Updates the implementation of all VTokens and Comptroller market facet in Core Pool
+    2. Updates the implementation of VTokens and Comptrollers in IL
+    3. Accepts the ownership of the NativeTokenGateway contract
+    4. Gives call permissions to Timelocks for seizeVenus function
+    5. Gives call permissions to Timelocks for setLastRewardingBlocks function of RewardsDistributor
+    6. Executes seizeVenus
+    7. Sets XVS and vXVS address in unitroller`,
+    forDescription: "I agree that Venus Protocol should proceed with this proposal",
+    againstDescription: "I do not think that Venus Protocol should proceed with this proposal",
+    abstainDescription: "I am indifferent to whether Venus Protocol proceeds or not",
+  };
+
+  return makeProposal(
+    [
+      {
+        target: UNITROLLER,
+        signature: "_setPendingImplementation(address)",
+        params: [DIAMOND],
+      },
+      {
+        target: DIAMOND,
+        signature: "_become(address)",
+        params: [UNITROLLER],
+      },
+      {
+        target: UNITROLLER,
+        signature: "diamondCut((address,uint8,bytes4[])[])",
+        params: [params],
+      },
+      ...CORE_MARKETS.map(vToken => {
+        return {
+          target: vToken.address,
+          signature: "_setImplementation(address,bool,bytes)",
+          params: [NEW_VBEP20_DELEGATE_IMPL, false, "0x"],
+        };
+      }),
+      {
+        target: COMPTROLLER_BEACON,
+        signature: "upgradeTo(address)",
+        params: [NEW_COMPTROLLER_IMPLEMENTATION],
+      },
+      {
+        target: VTOKEN_BEACON,
+        signature: "upgradeTo(address)",
+        params: [NEW_VTOKEN_IMPLEMENTATION],
+      },
+      {
+        target: NATIVE_TOKEN_GATEWAY,
+        signature: "acceptOwnership()",
+        params: [],
+      },
+
+      ...[NORMAL_TIMELOCK, FAST_TRACK_TIMELOCK, CRITICAL_TIMELOCK].map((timelock: string) => ({
+        target: ACM,
+        signature: "giveCallPermission(address,string,address)",
+        params: [UNITROLLER, "seizeVenus(address[],address)", timelock],
+      })),
+
+      {
+        target: ACM,
+        signature: "giveCallPermission(address,string,address)",
+        params: [ZERO_ADDRESS, "setLastRewardingBlocks(address[],uint32[],uint32[])", NORMAL_TIMELOCK],
+      },
+
+      {
+        target: UNITROLLER,
+        signature: "_setXVSToken(address)",
+        params: [XVS],
+      },
+      {
+        target: UNITROLLER,
+        signature: "_setXVSVToken(address)",
+        params: [XVSVTOKEN],
+      },
+      {
+        target: UNITROLLER,
+        signature: "seizeVenus(address[],address)",
+        params: [accounts1, UNITROLLER],
+      },
+      {
+        target: UNITROLLER,
+        signature: "seizeVenus(address[],address)",
+        params: [accounts2, UNITROLLER],
+      },
```

Comment:
Maybe move this to bscmainnet2.ts so that we have a clear separation between implementation changes and administrative actions?

Reply:
We can not move these commands to bscmainnet2 vip as if we move these commands there then bscmainnet2 vip goes out of gas as in each command 3 accounts are included and for each account the seizeVenus function first updates the rewards for each market then it calculates the total accruedVenus. Due to this big tx the vip does not execute. That's why i kept these two commands with accounts with highest **accruedVenus** here.